build_tools/scripts/compute_common_version.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This scripts grabs the X.Y.Z[.dev]` version identifier from the
+# compiler's and runtime's `version.json` files and computes the
+# shared version.
+
+import argparse
+from pathlib import Path
+import json
+from datetime import datetime
+import subprocess
+
+from packaging.version import Version
+
+
+parser = argparse.ArgumentParser()
+release_type = parser.add_mutually_exclusive_group()
+release_type.add_argument("-rc", "--nightly-release", action="store_true")
+release_type.add_argument("-dev", "--development-release", action="store_true")
+release_type.add_argument("--custom-string", action="store", type=str)
+args = parser.parse_args()
+
+THIS_DIR = Path(__file__).parent.resolve()
+
+VERSION_FILE_COMPILER = THIS_DIR / "../../compiler/version.json"
+VERSION_FILE_RUNTIME = THIS_DIR / "../../runtime/version.json"
```

Comment:
nit: could restructure this
```suggestion
THIS_DIR = Path(__file__).parent.resolve()
REPO_ROOT = THIS_DIR.parent.parent

VERSION_FILE_COMPILER = REPO_ROOT / "compiler/version.json"
VERSION_FILE_RUNTIME = REPO_ROOT / "runtime/version.json"
```